recipe/post-link.bat

```diff
@@ -2,27 +2,57 @@
 setlocal ENABLEDELAYEDEXPANSION
 
 set menudir=%PREFIX%\Menu
+set menu=%menudir%\spyder-menu.json
+set logfile=%PREFIX%\.messages.txt
 
+rem  Check for conda-based install
 if exist "%menudir%\conda-based-app" (
-    :: Installed in installer environment, abridge shortcut name
+    rem  Abridge shortcut name
     call :patch " ^({{ ENV_NAME }}^)="
+
+    rem  Nothing more to do for conda-based install
+    goto :exit
 )
 
-if exist "%PREFIX%\condabin\" if exist "%PREFIX%\envs\" (
-    :: Installed in a base environment, use distribution name
-    call :patch "ENV_NAME=DISTRIBUTION_NAME"
+rem  Check for CONDA_PYTHON_EXE
+if not exist "%conda_python_exe%" (
+    rem  CONDA_PYTHON_EXE environment variable does not exist.
+    rem  v1 type shortcuts will not work
+    goto :base_env
 )
 
+rem  Check menuinst version
+for /F "tokens=*" %%i in (
+    '%conda_python_exe% -c "from importlib.metadata import version; from packaging.version import parse; print(parse(version('menuinst')) < parse('2'))"'
```

Comment:
`importlib_metadata` or `packaging` could not be available in the base env. So, I think it's better to use only standard library modules here.